components/feature/qr/src/main/java/mozilla/components/feature/qr/QrFragment.kt

```diff
@@ -211,6 +224,7 @@ class QrFragment : Fragment() {
     }
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        coroutineScope = view.toScope()
```

Comment:
We used to decode bitmaps on a background thread and now this is happening on the main thread. In case decoding is slow, we may introduce ANRs. It's fast on my devices here, but wondering if should switch to a different scope? Wdyt?

Reply:
`onImageAvailable` still comes on a background thread.
I used the `coroutineScope` scoped to Fragment's View's scope just to have the benefit of structured concurrency.
Though if we cancel the coroutine either way in onStop we'll not really need it...
Will create a new Dispatchers.Default scope and use that just in `launch`. Should help on simplifying the code and tests, thanks!